File: problems_0-1000/9-PalindromeNumber/java/Solution.java

public class Solution {
    
    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.isPalindrome(2332));
    }

    public boolean isPalindrome(int x) {
        if (x < 0 || (x % 10 == 0 && x != 0)) {
            return false;
        }
        
        int reversedHalf = 0;
        while (x > reversedHalf) {
            reversedHalf = reversedHalf * 10 + x % 10;
            x = x / 10;
        }

        return x == reversedHalf || x == reversedHalf / 10;

    }
}

File: problems_0-1000/9-PalindromeNumber/python/solution.py

class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0 or (x != 0 and x % 10 == 0):
            return False

        reversedHalf = 0
        while x > reversedHalf:
            reversedHalf = reversedHalf * 10 + x % 10
            x = x // 10
        return x == reversedHalf or x == reversedHalf // 10

if __name__ == '__main__':
    x = 34543
    print(Solution().isPalindrome(x))